int32 id
float32[3] pose_r
float32[3] pose_t